docs/en/sql-reference/functions/splitting-merging-functions.md

```diff
@@ -91,20 +91,21 @@ SELECT splitByString('', 'abcde');
 └────────────────────────────┘
 ```
 
-## splitByRegexp(regexp, s)
+## splitByRegexp(regexp, s[, max_tokens])
 
-Splits a string into substrings separated by a regular expression. It uses a regular expression string `regexp` as the separator. If the `regexp` is empty, it will split the string `s` into an array of single characters. If no match is found for this regular expression, the string `s` won't be split.
+Splits a string into substrings separated by a regular expression. It uses a regular expression string `regexp` as the separator. If the `regexp` is empty, it will split the string `s` into an array of single characters. If no match is found for this regular expression, the string `s` won't be split. `max_tokens` limit the number of returned tokens from function `splitByRegexp`.
 
 **Syntax**
 
 ``` sql
-splitByRegexp(regexp, s)
+splitByRegexp(regexp, s[, max_tokens])
 ```
 
 **Arguments**
 
 -   `regexp` — Regular expression. Constant. [String](../data-types/string.md) or [FixedString](../data-types/fixedstring.md).
 -   `s` — The string to split. [String](../../sql-reference/data-types/string.md).
+-   `max_tokens` — An optional `Int64` defaulting to 0. When `max_tokens` > 0, the returned tokens will be no more than `max_tokens`, otherwise `splitByRegexp` will return as many tokens as possible.
```

Comment:
Ah, I should have read the previous discussion in this PR ^^

Reply:
I also thought it is more reasonable to return empty arrays if input 0. But in spark function `split`, 0 means no limit. Since user can directly use empty string instead of  splitxxx(str, delim, 0),  can we just keep the same semantics with spark?

Reply:
I don't mind if we go for Spark or Python semantics. What would be nice is if all `splitBy*()` functions had the same semantics for their 3rd parameter. And since the 3rd parameter of `splitByChar()` isn't documented, we are free to change it, imho.